### PR TITLE
[Enhancement] [ALS-1861] [iOS] Update Auth SDK's AutHelper to support API key 

### PR DESCRIPTION
**Pull request by zeeshanmakeen**
By skipping any default authentication in case of API key and only use api key for authentication

ALS-1861

### CODE REVIEW

File: Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift [hunk 1]
```diff
@@ -22,29 +24,40 @@ import AwsCommonRuntimeKit
...
         return credentialProvider
     }
     
     public func authenticateWithCredentialsProvider(credentialsProvider: CredentialsProvider, region: String) async throws -> LocationCredentialsProvider? {
         let credentialProvider = LocationCredentialsProvider(credentialsProvider: credentialsProvider)
         credentialProvider.setRegion(region: region)
+        locationCredentialsProvider = credentialProvider
+        if let credentialsProvider = locationCredentialsProvider {
+            amazonLocationClient = AmazonLocationClient(locationCredentialsProvider: credentialsProvider)
+        }
```
Comment (mbalfour-amzn) [inline Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift]: Is this missing `try await amazonLocationClient?.initialiseLocationClient()` ?

File: Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift [hunk 2]
```diff
@@ -22,29 +24,40 @@ import AwsCommonRuntimeKit
...
-        guard let locationCredentialsProvider = self.locationCredentialsProvider else {
-            return nil
-        }
-        return AmazonLocationClient(locationCredentialsProvider: locationCredentialsProvider)
+        return amazonLocationClient
+    }
+    
+    public func getLocationClient() -> LocationClient?
+    {
+        return amazonLocationClient?.locationClient
```
Comment (mbalfour-amzn) [inline Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift]: This is changing the public API so that getLocationClient() now returns LocationClient instead of AmazonLocationClient. It seems like this would break any existing code. Why is this change necessary?

Comment (mbalfour-amzn) [reply Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift]: (For example see here - https://github.com/aws-geospatial/amazon-location-samples-ios/blob/main/quick-start/QuickStartTrackingApp/ViewModels/TrackingViewModel.swift#L85 )

Looking deeper into the auth helper, I think I understand why you want to expose the underlying location client, since I'm not sure how you would use AmazonLocationClient without extending it. But I think we need to do this in a way that wouldn't break any existing code.

File: Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift (hunk 1 above)
Comment (zeeshanmakeen) [reply Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift]: Thank you, I have added the initialiseLocationClient and updated the test case

File: Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift (hunk 2 above)
Comment (zeeshanmakeen) [reply Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift]: Yes you are right. I did this change to make it simpler. But I have reverted it to the previous function signature so it won't break the code.

**Review by mbalfour-amzn — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]
